app/src/main/java/com/arctouch/codechallenge/movies_list/MoviesListActivity.java

```diff
@@ -0,0 +1,300 @@
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.design.widget.CoordinatorLayout;
+import android.support.design.widget.Snackbar;
+import android.support.v4.widget.SwipeRefreshLayout;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.arctouch.codechallenge.R;
+import com.arctouch.codechallenge.data.Cache;
+import com.arctouch.codechallenge.listener.EndlessRecyclerViewScrollListener;
+import com.arctouch.codechallenge.listener.IMovieClickListener;
+import com.arctouch.codechallenge.listener.ISearchViewTextSubmitListener;
+import com.arctouch.codechallenge.listener.ISnackBarActionListener;
+import com.arctouch.codechallenge.model.Genre;
+import com.arctouch.codechallenge.model.Movie;
+import com.arctouch.codechallenge.movie_details.MovieDetailsActivity;
+import com.arctouch.codechallenge.movies_list.presentation.MoviesListContract;
+import com.arctouch.codechallenge.movies_list.presentation.MoviesListPresenter;
+import com.arctouch.codechallenge.util.MaterialSearchViewHelper;
+import com.arctouch.codechallenge.util.NetworkUtils;
+import com.arctouch.codechallenge.util.SnackUtils;
+import com.miguelcatalan.materialsearchview.MaterialSearchView;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+public class MoviesListActivity extends AppCompatActivity implements MoviesListContract.View, IMovieClickListener, ISearchViewTextSubmitListener, ISnackBarActionListener {
+
+    @BindView(R.id.coordinator_main)
+    CoordinatorLayout coordinatorLayout;
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @BindView(R.id.search_view)
+    MaterialSearchView searchView;
+
+    @BindView(R.id.swipe_refresh)
+    SwipeRefreshLayout swipeRefresh;
+
+    @BindView(R.id.recyclerView)
+    RecyclerView recyclerMovies;
+
+    private MoviesListPresenter moviesListPresenter;
+    private MoviesListAdapter adapter;
+    private ArrayList<Movie> movieList;
+    private EndlessRecyclerViewScrollListener endlessRecyclerViewScrollListener;
+    private String searchedName;
+    private int currentPage;
+    private Menu menu;
+    private Bundle args;
+
+    private static final String BUNDLE_SEARCHED_NAME = "search_by_name";
+    private static final String BUNDLE_CURRENT_PAGE = "current_page";
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
```

Comment:
Activity's `onCreate` methods can get pretty long. What are some of the techniques you can use to make sure the activity's set up code remains readable?

Reply:
There is one simple technique which is just for code readability purposes that is creating like a setup method that contains the initializations, like `Bundle`, `Arrays`, etc, so after this `setup` method would have the method calls and others. Of course, if it is seen that the `onCreate` and the Activity itself is growing too much, there is a need to separate things because the cohesion would be lost.